### PR TITLE
PXB-1745: Introduce build script

### PR DESCRIPTION
**Pull request by Sudokamikaze**
TESTED:
* Bin_tar
* Src_tar
* Centos 6/7/8 srpm & rpm
* Debian 9/10 sdeb & deb
* Ubuntu testing

### CODE REVIEW

**Comment by it-percona** [comment]
[CLA assistant check](https://cla-assistant.percona.com:443/percona/percona-xtrabackup?pullRequest=858) <br/>All committers have signed the CLA.

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 1]
```diff
@@ -0,0 +1,587 @@
...
+            exit 1
+        fi
+    fi
+    return
+}
+
+add_percona_yum_repo(){
+    if [ ! -f /etc/yum.repos.d/percona-dev.repo ]; then
+        curl -o /etc/yum.repos.d/percona-dev.repo https://jenkins.percona.com/yum-repo/percona-dev.repo
+	    sed -i 's:$basearch:x86_64:g' /etc/yum.repos.d/percona-dev.repo
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: indentation

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 2]
```diff
@@ -0,0 +1,587 @@
...
+    fi
+    return
+}
+
+add_percona_apt_repo(){
+    if [ ! -f /etc/apt/sources.list.d/percona-dev.list ]; then
+        echo "deb http://jenkins.percona.com/apt-repo/ $OS_NAME main" > /etc/apt/sources.list.d/percona-dev.list
+        wget -q -O - http://jenkins.percona.com/apt-repo/8507EFA5.pub | apt-key add -
+  	    wget -q -O - http://jenkins.percona.com/apt-repo/CD2EFD2A.pub | apt-key add -
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: indentation

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 3]
```diff
@@ -0,0 +1,587 @@
...
+    fi
+    return
+}
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: please check if we really install any packages from this repo. for example this function was removed in many build scripts

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 4]
```diff
@@ -0,0 +1,587 @@
...
+
+    cd ${PXBDIR}
+    sed -i "s:@@XB_VERSION_MAJOR@@:${XB_VERSION_MAJOR}:g" storage/innobase/xtrabackup/utils/percona-xtrabackup.spec
+    sed -i "s:@@XB_VERSION_MINOR@@:${XB_VERSION_MINOR}:g" storage/innobase/xtrabackup/utils/percona-xtrabackup.spec
+    sed -i "s:@@XB_VERSION_PATCH@@:${XB_VERSION_PATCH}:g" storage/innobase/xtrabackup/utils/percona-xtrabackup.spec
+    #
+    if [ -z "${XB_VERSION_EXTRA}" ]; then
+    EXTRAVER="%{nil}"
+    #  RPM_EXTRAVER="%{nil}"
+    RPM_EXTRAVER=${RPM_RELEASE}
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: indentation

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 5]
```diff
@@ -0,0 +1,587 @@
...
+    else 
+    EXTRAVER=${XB_VERSION_EXTRA}
+    #  RPM_EXTRAVER=.${XB_VERSION_EXTRA#-}
+    RPM_EXTRAVER=${XB_VERSION_EXTRA#-}.${RPM_RELEASE}
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: indentation

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 6]
```diff
@@ -0,0 +1,587 @@
...
+            PKGLIST+=" libcurl-devel cmake libaio-devel zlib-devel libev-devel bison make gcc"
+            PKGLIST+=" rpm-build libgcrypt-devel ncurses-devel readline-devel openssl-devel gcc-c++"
+            PKGLIST+=" install vim-common rpmlint"
+            until yum -y install ${PKGLIST}; do
+                echo "waiting"
+                sleep 1
+            done
+            /usr/bin/pip3 install sphinx
+        else
+            until yum -y install epel-release centos-release-scl; do
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh:230]: epel-release is available on all centos system. so we can  install it everywhere

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 7]
```diff
@@ -0,0 +1,587 @@
...
+                echo "waiting"
+                sleep 1
+            done
+            # Workaround for Centos > 7 Sphinx
+            if [[ "${RHEL}" -lt 7 ]]; then
+                /usr/bin/pip3 install sphinx
+            fi
+        fi
+    else
+        apt-get -y install lsb-release gnupg2
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: use gnupg

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 8]
```diff
@@ -0,0 +1,587 @@
...
+        PKGLIST+=" bison cmake devscripts debconf debhelper automake bison ca-certificates libcurl4-openssl-dev"
+        PKGLIST+=" cmake debhelper libaio-dev libncurses-dev libtool libz-dev"
+        PKGLIST+=" libgcrypt-dev libev-dev lsb-release python-sphinx python-docutils"
+        PKGLIST+=" build-essential rsync libdbd-mysql-perl libnuma1 socat"
+
+        if [ "${OS_NAME}" = "cosmic" ]; then
+            PKGLIST+=" libssl1.0-dev"
+        else
+            PKGLIST+=" libssl-dev"
+        fi
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: we can remove this block. cosmic is EOL
just install needed package

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 9]
```diff
@@ -0,0 +1,587 @@
...
+        GCRYPT_VER_TMP=$(dpkg -l|grep -i libgcrypt | grep -v dev | awk '{print $2}' | awk -F':' '{print $1}')
+        export GCRYPT_VER=".${GCRYPT_VER_TMP}"
+    fi
+
+    rm -fr TARGET && mkdir TARGET
+    rm -fr ${TARFILE%.tar.gz}
+    tar xzf ${TARFILE}
+    cd ${TARFILE%.tar.gz}
+    #
+    sed -i 's|.tar.gz|.${OS_RELEASE}.tar.gz|' ./storage/innobase/xtrabackup/utils/build-binary.sh
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: please also add libgcrypt version i.e ${OS_RELEASE}$GCRYPT_VER{}

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh (hunk 1 above)
Comment (Sudokamikaze) [reply storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: Done

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh (hunk 4 above)
Comment (Sudokamikaze) [reply storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: Done

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh (hunk 5 above)
Comment (Sudokamikaze) [reply storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: Done

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh (hunk 7 above)
Comment (Sudokamikaze) [reply storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: Done

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh (hunk 8 above)
Comment (Sudokamikaze) [reply storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: Done

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh (hunk 9 above)
Comment (Sudokamikaze) [reply storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: Done

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh (hunk 3 above)
Comment (Sudokamikaze) [reply storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: Done, removed

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh (hunk 2 above)
Comment (Sudokamikaze) [reply storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: Done

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh (hunk 6 above)
Comment (Sudokamikaze) [reply storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh:230]: It is. But there are no EPEL packages installed for Cent8, so I decided to keep it for >8 versions

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 10]
```diff
@@ -0,0 +1,585 @@
...
+    git reset --hard
+    #
+    source XB_VERSION
+    cat XB_VERSION > ../percona-xtrabackup-8.0.properties
+    echo "REVISION=${REVISION}" >> ../percona-xtrabackup-8.0.properties
+    BRANCH_NAME="${BRANCH}"
+    echo "BRANCH_NAME=$(echo ${BRANCH_NAME} | awk -F '/' '{print $(NF)}')" >> ../percona-xtrabackup-8.0.properties
+    echo "PRODUCT=Percona-XtraBackup" >> ../percona-xtrabackup-8.0.properties
+    echo "PRODUCT_FULL=Percona-XtraBackup-${XB_VERSION_MAJOR}.${XB_VERSION_MINOR}-${XB_VERSION_PATCH}${XB_VERSION_EXTRA}" >> ../percona-xtrabackup-8.0.properties
+    echo "PRODUCT_UL_DIR=XtraBackup" >> ../percona-xtrabackup-8.0.properties
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: please use `Percona-XtraBackup-8.0`

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 11]
```diff
@@ -0,0 +1,585 @@
...
+                pip install sphinx
+            fi
+        fi
+    else
+        apt-get -y install lsb-release gnupg git wget
+
+        until apt-get update; do
+    	    sleep 1
+            echo "waiting"
+        done
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: why do we need it? we didn't install any new repos

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 12]
```diff
@@ -0,0 +1,585 @@
...
+
+        PKGLIST+=" bison cmake devscripts debconf debhelper automake bison ca-certificates libcurl4-openssl-dev"
+        PKGLIST+=" cmake debhelper libaio-dev libncurses-dev libtool libz-dev"
+        PKGLIST+=" libgcrypt-dev libev-dev lsb-release python-sphinx python-docutils"
+        PKGLIST+=" build-essential rsync libdbd-mysql-perl libnuma1 socat libssl-dev"
+        
+        if [ "${OS_NAME}" == "jessie" ]; then
+            PKGLIST+=" libreadline6"
+        fi
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: we do not build it for jessie.

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]
also waiting for jenkins job

**Comment by Sudokamikaze** [comment]
Made draft as requested.

Will be available after Jenkins testing

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 13]
```diff
@@ -0,0 +1,589 @@
...
+        exit 1
+    fi
+    for file in 'dsc' 'orig.tar.gz' 'debian.tar.*' 'changes'
+    do
+        get_deb_sources $file
+    done
+    cd $WORKDIR
+
+    echo "DEB_RELEASE=${DEB_RELEASE}" >> percona-xtrabackup-8.0.properties
+    DEBIAN_VERSION="$(lsb_release -sc)"
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: you should already have value in ${OS_NAME}

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 14]
```diff
@@ -0,0 +1,589 @@
...
+build_tarball(){
+    if [ $TARBALL = 0 ]
+    then
+        echo "Binary tarball will not be created"
+        return;
+    fi
+    get_tar "source_tarball"
+    cd $WORKDIR
+    TARFILE=$(basename $(find . -name 'percona-xtrabackup-*.tar.gz' | sort | tail -n1))
+    if [ -f /etc/debian_version ]; then
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: get_system?

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 15]
```diff
@@ -0,0 +1,589 @@
...
+    SHORTVER=$(echo ${VERSION} | awk -F '.' '{print $1"."$2}')
+    TMPREL=$(echo ${TARFILE}| awk -F '-' '{print $4}')
+    RELEASE=${TMPREL%.tar.gz}
+    if [ -f /etc/redhat-release ]; then
+        GCRYPT_VER_TMP=$(yum list installed|grep -i libgcrypt | grep -v dev|head -n1|awk '{print $2}'|awk -F "-" '{print $1}'|sed 's/\.//g'|sed 's/[a-z]$//')
+        export GCRYPT_VER=".libgcrypt${GCRYPT_VER_TMP}"
+    elif [ -f /etc/debian_version ]; then
+        GCRYPT_VER_TMP=$(dpkg -l|grep -i libgcrypt | grep -v dev | awk '{print $2}' | awk -F':' '{print $1}')
+        export GCRYPT_VER=".${GCRYPT_VER_TMP}"
+    fi
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: this block can be improved based on the info from get_system subroutine

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh (hunk 13 above)
Comment (Sudokamikaze) [reply storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: Done

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh (hunk 14 above)
Comment (Sudokamikaze) [reply storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: Done

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh (hunk 15 above)
Comment (Sudokamikaze) [reply storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: Done

**Comment by Sudokamikaze** [comment]
@EvgeniyPatlan recent changes are also tested in jenkins

File: storage/innobase/xtrabackup/utils/debian/control [hunk 16]
```diff
@@ -17,8 +17,8 @@ Build-Depends: automake,
                libev-dev,
                libcurl-dev,
                lsb-release,
-               python-sphinx (>= 1.0.1),
-               python-docutils (>= 0.6)
+               python-sphinx (>= 1.0.1) | python3-sphinx (>= 1.8.5),
+               python-docutils (>= 0.6) | python3-docutils (>= 0.16)
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/debian/control:21]: are you sure that minimal needed versions for python3* are correct?

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 17]
```diff
@@ -0,0 +1,585 @@
+#!/bin/sh
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: #!/usr/bin/env bash

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 18]
```diff
@@ -0,0 +1,585 @@
...
+    if [ $retval != 0 ]
+    then
+        echo "There were some issues during repo cloning from github. Please retry one more time"
+        exit 1
+    fi
+
+    cd percona-xtrabackup
+    if [ ! -z $BRANCH ]
+    then
+        git checkout $BRANCH
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh:112]: please add before this line
        git reset --hard
        git clean -xdf

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 19]
```diff
@@ -0,0 +1,585 @@
...
+        EXTRAVER=${XB_VERSION_EXTRA}
+        RPM_EXTRAVER=${XB_VERSION_EXTRA#-}.${RPM_RELEASE}
+    fi
+    #
+    sed -i "s:@@XB_VERSION_EXTRA@@:${EXTRAVER}:g" storage/innobase/xtrabackup/utils/percona-xtrabackup.spec
+    sed -i "s:@@XB_RPM_VERSION_EXTRA@@:${RPM_EXTRAVER}:g" storage/innobase/xtrabackup/utils/percona-xtrabackup.spec
+    sed -i "s:@@XB_REVISION@@:${REVISION}:g" storage/innobase/xtrabackup/utils/percona-xtrabackup.spec
+    #
+    #for 8.0 alpha release
+    sed -i "s:= %{version}-%{release}:> 8.0:g" storage/innobase/xtrabackup/utils/percona-xtrabackup.spec
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: this is not needed any more

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 20]
```diff
@@ -0,0 +1,585 @@
...
+            if [[ "${RHEL}" -eq 7 ]]; then
+                /usr/bin/pip3 install sphinx
+            elif [[ "${RHEL}" -eq 6 ]]; then
+                source /opt/rh/rh-python36/enable
+                pip install sphinx
+            fi
+        fi
+    else
+        apt-get update
+        apt-get -y install lsb-release gnupg git wget
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: DEBIAN_FRONTEND=noninteractive

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 21]
```diff
@@ -0,0 +1,585 @@
...
+        return;
+    fi
+    if [ "x$OS" = "xdeb" ]
+    then
+        echo "It is not possible to build src rpm here"
+        exit 1
+    fi
+    cd $WORKDIR
+    get_tar "source_tarball"
+    rm -fr rpmbuild
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: if [ -d rpmbuild ]; then
    rm -fr rpmbuild
fi

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 22]
```diff
@@ -0,0 +1,585 @@
...
+    if [ "$OS" == "rpm" ]; then
+        if [ "${RHEL}" -eq 8 ]; then
+            export PATH="/usr/local/bin:$PATH"
+        elif [ "${RHEL}" -eq 7 ]; then
+            source /opt/rh/devtoolset-7/enable
+        elif [ "${RHEL}" -eq 6 ]; then
+            source /opt/rh/devtoolset-7/enable
+            source /opt/rh/rh-python36/enable
+        fi
+    fi
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: As this code is used in several places it is better to create separate function and just call it when needed

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]

File: storage/innobase/xtrabackup/utils/debian/control (hunk 16 above)
Comment (Sudokamikaze) [reply storage/innobase/xtrabackup/utils/debian/control:21]: It's the only available versions:
```
root@ubuntu-20:/home/vagrant# apt policy python3-sphinx
python3-sphinx:
  Installed: (none)
  Candidate: 1.8.5-7ubuntu3
  Version table:
     1.8.5-7ubuntu3 500
        500 http://archive.ubuntu.com/ubuntu focal/main amd64 Packages
        500 http://archive.ubuntu.com/ubuntu focal/main i386 Packages

root@ubuntu-20:/home/vagrant# apt policy python3-docutils
python3-docutils:
  Installed: (none)
  Candidate: 0.16+dfsg-2
  Version table:
     0.16+dfsg-2 500
        500 http://archive.ubuntu.com/ubuntu focal/main amd64 Packages
        500 http://archive.ubuntu.com/ubuntu focal/main i386 Packages
```

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh (hunk 22 above)
Comment (Sudokamikaze) [reply storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: Also added CMAKE handling to new function

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh (hunk 19 above)
Comment (Sudokamikaze) [reply storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: Removed

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 23]
```diff
@@ -0,0 +1,575 @@
...
+    cp rpmbuild/RPMS/*/*.rpm ${CURDIR}/rpm    
+}
+
+build_source_deb(){
+    if [ $SDEB = 0 ]
+    then
+        echo "source deb package will not be created"
+        return;
+    fi
+    if [ "x$OS" = "xrmp" ]
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: rpm

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 24]
```diff
@@ -0,0 +1,575 @@
...
+    
+}
+
+build_deb(){
+    if [ $DEB = 0 ]
+    then
+        echo "source deb package will not be created"
+        return;
+    fi
+    if [ "x$OS" = "xrmp" ]
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: rpm

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 25]
```diff
@@ -0,0 +1,575 @@
...
+            done
+            PKGLIST+=" devtoolset-7-gcc-c++ devtoolset-7-binutils"
+            PKGLIST+=" devtoolset-7-libasan-devel devtoolset-7-libubsan-devel"
+            PKGLIST+=" devtoolset-7-valgrind devtoolset-7-valgrind-devel"
+            PKGLIST+=" wget libcurl-devel cmake cmake3 make gcc gcc-c++ libev-devel openssl-devel rpm-build"
+            PKGLIST+=" libaio-devel perl-DBD-MySQL vim-common ncurses-devel readline-devel readline"
+            PKGLIST+=" zlib-devel libgcrypt-devel bison"
+            PKGLIST+=" socat numactl"
+            if [[ "${RHEL}" -eq 7 ]]; then
+                PKGLIST+=" numactl-libs perl-Digest-MD5  python3-pip python3-docutils python3-setuptools python3-wheel python3-docutils"
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: No package python3-docutils available.
No package python3-docutils available.

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh [hunk 26]
```diff
@@ -0,0 +1,575 @@
...
+                sleep 1
+            done
+            /usr/bin/pip3 install sphinx
+        else
+            until yum -y install epel-release centos-release-scl; do
+                yum clean all
+                sleep 1
+                echo "waiting"
+            done
+            until yum -y makecache; do
```
Comment (EvgeniyPatlan) [inline storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh:235]: (6/22): base/7/x86_64/other_db                                                                                  | 2.6 MB  00:00:01     
epel/x86_64/updateinfo         FAILED                                                                ] 5.1 MB/s |  17 MB  00:00:06 ETA 
https://ftp.icm.edu.pl/pub/Linux/fedora/linux/epel/7/x86_64/repodata/ea5cfe85e50491bfba7524d64236cd8a954f1b59e82813857e2f2e90d5de5a45-updateinfo.xml.bz2: [Errno 14] HTTPS Error 404 - Not Found
Trying other mirror.
To address this issue please refer to the below wiki article 

https://wiki.centos.org/yum-errors

If above article doesn't help to resolve this issue please use https://bugs.centos.org/.

(7/22): centos-sclo-rh/x86_64/other_db                                                                          | 1.3 MB  00:00:02     
epel/x86_64/prestodelta        FAILED                                          
http://epel.besthosting.ua/7/x86_64/repodata/bed3cebba0d213f674b0df99ed8d2fcf4fd2204fcc1fb880b294cb9181ba217c-prestodelta.xml.gz: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.

do we really need this?

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh (hunk 26 above)
Comment (Sudokamikaze) [reply storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh:235]: Yes we do:
`devtoolset-7-gcc` and dependency are being installed from this repository. Plus, `rh-python` packages are from this repo as well.

File: storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh (hunk 25 above)
Comment (Sudokamikaze) [reply storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh]: Removed as it populates from `pip`

**Review by EvgeniyPatlan — APPROVED** [review]

**Review by vorsel — APPROVED** [review]

**Pull request merged**